apollo-federation/src/sources/connect/json_selection/README.md

```diff
@@ -82,9 +82,10 @@ below.
 
 ```ebnf
 JSONSelection        ::= PathSelection | NamedSelection*
-SubSelection         ::= "{" NamedSelection* "}"
+SubSelection         ::= "{" OutputTypeAnnotation? NamedSelection* "}"
+OutputTypeAnnotation ::= "<" Identifier ">"
 NamedSelection       ::= NamedPathSelection | PathWithSubSelection | NamedFieldSelection | NamedGroupSelection
-NamedPathSelection   ::= Alias PathSelection
+NamedPathSelection   ::= (Alias | "...") PathSelection
```

Comment:
does this allow for future keywords like `if` or `match` directly after the `...`?

Reply:
If the `PathSelection` started with an `if` token, that would be a potential concern for making `if` a keyword in the future, but if we stick with the the idea of a parenthesized test (i.e. `if (test) ...`), the `(` character should prevent parsing as a `PathSelection`.

Same goes for `match (value) ...`, as long as we use parentheses, or as long as we're willing to make the `match` field illegal immediately after `...` (a breaking change but not unthinkable).